### PR TITLE
Syntactic sugar for hasBeenInterrupted

### PR DESCRIPTION
**Pull request by kesselb**
Some syntactic sugar for hasBeenInterrupted. In 8 out of 9 occurrences the method is used like below:

```
if ($this->hasBeenInterrupted()) {
    throw new InterruptedException();
}
```
Changed the method to throw an exception when current command has been interrupted. I guess the method is only used in nextcloud/server and nextcloud/fulltextsearch (@daita)

### CODE REVIEW

File: apps/files/lib/Command/ScanAppData.php [hunk 1]
```diff
@@ -102,33 +102,25 @@ protected function scanFiles($verbose, OutputInterface $output) {
...
+				$output->writeln('Error while scanning, storage not available (' . $e->getMessage() . ')');
 			});
 			# count only
 		} else {
 			$scanner->listen('\OC\Files\Utils\Scanner', 'scanFile', function () use ($output) {
 				$this->filesCounter += 1;
-				if ($this->hasBeenInterrupted()) {
-					throw new InterruptedException();
-				}
+				$this->hasBeenInterrupted();
```
Comment (ChristophWurst) [inline apps/files/lib/Command/ScanAppData.php]: IMO the name isn't ideal as it still sounds like a *getter*. How about something like `abortIfInterrupted`? That would make it clear that the method exits the current method on certain conditions.

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
I like the improvement a lot, but the naming isn't idea :wink:

File: apps/files/lib/Command/ScanAppData.php (hunk 1 above)
Comment (ChristophWurst) [reply apps/files/lib/Command/ScanAppData.php]: Alternative: `ensureNotInterrupted`.

Comment (danielkesselberg) [reply apps/files/lib/Command/ScanAppData.php]: Thank you :+1: `abortIfInterrupted` is more clear.

**Review by ChristophWurst — APPROVED** [review]
Code looks good!

**Review by rullzer — APPROVED** [review]

**Pull request merged**